discussion/src/main/java/org/openedx/discussion/presentation/comments/DiscussionCommentsFragment.kt

```diff
@@ -391,7 +397,7 @@ private fun DiscussionCommentsScreen(
                                         )
                                     }
                                     if (uiState.commentsData.isNotEmpty()) {
-                                        item {
+                                        item(uiState.count) {
```

Comment:
This introduces a logical issue. In `LazyListScope.item`, the key should be a stable, unique identifier so Compose can preserve item identity across recompositions. `uiState.count` changes on every update, which can cause state loss and unnecessary rebinds. We don’t need this change; the UI already updates via `commentCount++` in the `ViewModel` (which flows into `uiState`). Let’s revert this change. 🙌🏻